memory.tex

```diff
@@ -0,0 +1,1939 @@
+of memory, so it is often convenient for a program to set the memory
+configuration at the beginning of a task and use 16-bit addresses.
+
+Address translation is a common feature of most microcomputers. The Commodore
+64 uses banking to allow programs to access 64KB of RAM {\em and} additional
+ROM and I/O registers using only 16-bit addresses. The Commmodore 65 was
+designed such that the memory system could be configured to resemble a
+Commodore 64 memory map through 16-bit address translation, and also support
+C64-style remapping of ROM and I/O registers.
+
+Overall, the MEGA65 provides multiple mechanisms of address translation. Memory
+system configuration is one of the more complex topics of MEGA65 programming,
+and this chapter will cover it in detail.
+
+\subsection{Banks and Pages}
+
+MEGA65 addresses can be divided into regions of 64KB known as {\em banks}. It's
+easy to identify the bank given an address specified in hexadecimal: the bank
+number is the leftmost digits of the address, and the rightmost four digits are
+a location within the bank.
+
+\medskip
+\begin{center}
+\begin{tabular}{m{0.4cm}m{0.4cm}m{0.05cm}m{0.4cm}m{0.4cm}m{0.4cm}m{0.4cm}}
+\multicolumn{1}{c}{\huge\texttt{\$}} &
+\multicolumn{1}{c}{\huge\texttt{3}} &
+\multicolumn{1}{c}{ } &
+\multicolumn{1}{c}{\huge\texttt{F}} &
+\multicolumn{1}{c}{\huge\texttt{D}} &
+\multicolumn{1}{c}{\huge\texttt{0}} &
+\multicolumn{1}{c}{\huge\texttt{8}} \\
+\cline{2-2}\cline{4-5}
+& \multicolumn{1}{c}{\small Bank} & & \multicolumn{2}{c}{\small Page} &
+\multicolumn{2}{c}{ } \\
+\end{tabular}
+\end{center}
+\medskip
+
+For example, the address 3.FD08 (or \$003FD08) is an address in bank \$3. The
+address 0.1800 (\$0001800) is an address in bank \$0. The address 880.12AB
+(\$88012AB) is in bank \$880.\footnote{In a few cases, this is better understood as
+"megabyte \$88, bank \$0, offset \$12AB."}
+
+It is important to remember that if you see a 16-bit
+address such as \$2100, the actual bank that it refers to may not be bank 0.
+The full address depends on how the memory system is configured, and how the
+memory is accessed.
+
+A bank can be further divided into regions of 256 bytes known as {\em pages}.
+As with banks, the page is easy to identify from a hexadecimal address: it is
+the fourth and third rightmost digits. For example, page \$1C consists of
+\$1C00 -- \$1CFF. The address \$1CB2 is in page \$1C.
+
+\subsection{How Addresses are Stored in Memory}
+
+Computers often manipulate memory addresses as data, including storing
+addresses in RAM or in registers. A 16-bit address is stored as two bytes,
+and a 28-bit address is stored as four bytes (with the four leftmost bits set
+to zero).
+
+To determine the bytes for an address, take the hexadecimal digits and
+group them in pairs, starting from the {\em rightmost} pair. Each pair of
+digits is a byte value. For example, the address \$853FD08 can be represented
+by the bytes \$08, \$FD, \$03, and \$85.
```

Comment:
This doesn't match with the value above, `853FD08`. I suppose one should be changed so they match.
```suggestion
by the bytes \$08, \$FD, \$53, and \$08.
```

Reply:
Fixed and also updated the example address throughout to avoid repeating $08.